.travis.yml

```diff
@@ -11,20 +11,47 @@ cache:
     - $HOME/.composer/cache
 
 before_install:
-  - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || echo "xdebug not available"
-  - composer self-update
+  - |
+    if [ "x$COVERAGE" != "xyes" ]; then
+      mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || true
+    fi
+  - travis_retry composer self-update
 
 install: travis_retry composer update --prefer-dist
 
+before_script:
+  - |
+    if [ "x$COVERAGE" == "xyes" ] && [[ ! $(php -m | grep -si xdebug) ]]; then
+      echo "xdebug is required for coverage"
+      exit 1
```

Comment:
using `exit` in a command put in the .travis.yml will exit the whole build script, which will break the Travis output (as reporting for this command will not be done, as well as following tasks).

Reply:
First time I hear about it. :thinking: What would you suggest as replacement here?

Reply:
Well, there used to be a warning in the Travis doc about it in the past, but I cannot find it anymore (they reorganized their doc in the meantime).

I thinkk you can use `travis_terminate 1` for that (which is a command which will exit with `1` as exit code, and so become a failure).